topology/base.py

```diff
@@ -115,25 +126,36 @@ def _create_graph(
                 )
                 sinks.append(sink_node)
 
+                graph.add_node(sink_node)
+                sink = [sink for sink in sinks_cfg if sink['id'] == upstream_sink_id][0]
+                graph.nodes[sink_node]['a_x'] = sink['a_x']
+                graph.nodes[sink_node]['a_y'] = sink['a_y']
+
                 conveyors_sections[int(conveyor_id)].append(sink_node)
             elif upstream_cfg['type'] == 'conveyor':
                 upstream_conveyor_id = upstream_cfg['id']
                 upstream_conveyor_position = upstream_cfg['pos']
 
-                conveyors_sections[int(conveyor_id)].append(Section(
+                end_junction = Section(
                     type='junction', id=junction_idx, position=length
-                ))
+                )
 
-                conveyors_sections[upstream_conveyor_id].append(Section(
+                start_junction = Section(
                     type='junction', id=junction_idx, position=upstream_conveyor_position
-                ))
-
-                graph.add_node(
-                    Section(
-                        type='junction', id=junction_idx, position=upstream_conveyor_position
-                    )
                 )
 
+                conveyors_sections[int(conveyor_id)].append(end_junction)
+
+                conveyors_sections[upstream_conveyor_id].append(start_junction)
+
+                graph.add_node(start_junction)
```

Comment:
It is strange. Before we added one node and right now you add here two nodes. Could you please clarify these changes? It is unclear to me.

Reply:
You can add nodes using `add_node` or you can add two nodes using `add_edge` if nodes in this `edge` don't exist. Before we added this node using `add_edge` later. Now we add it before `add_edge` and `add_edge` simply add connection later.